### ISSUE TITLE
Support for plot composition (patchwork)

### ISSUE THREAD

**Issue by krassowski** [has2k1/plotnine#457]
[Patchwork](https://github.com/thomasp85/patchwork) has become an invaluable tool for ggplot2. It allows to compose plots with a simple grammar (using +, /, | operators).

How feasible would be implementing a similar functionality for plotnine? Would this be in scope?

![Screenshot from 2020-11-14 11-28-46](https://user-images.githubusercontent.com/5832902/99146078-9a2e9680-266c-11eb-96fa-c2acdb0dbb23.png)

**Comment by has2k1** [comment]
This is hard and may not be possible. ggplot2 builds on top of a system where plots are made then arranged. plotnine builds on top of matplotlib where space is allocated and plots are drawn into the spaces. The problem with the later is that space is allocated for the panels/facets/subplots but the objects (e.g axis ticks, axis labels/titles, titles, subtitles, associated legends, ...) that adorn the panels are not part of that calculation (and they cannot be because their dimensions are know only after they have been drawn). So you may have to check for overlapping objects and make adjustments.

This is what makes plot composition hard. However, `patchwork` has already had some [influence](https://matplotlib.org/3.3.0/tutorials/provisional/mosaic.html) but this while also limited still not easy to adapt for plotnine.

This issue is a in part duplicate of #46 which I try every now and then to reconsider but have so far failed.

**Comment by dhimmel** [comment]
Noting these two libraries mentioned in https://github.com/has2k1/plotnine/issues/46:

- https://github.com/benjaminleroy/cowpatch
- https://github.com/ponnhide/patchworklib
